useful-scripts/importer-publisher-tests/index.ts

```diff
@@ -0,0 +1,365 @@
+        Authorization: `Bearer ${JWT_TOKEN}`,
+        'Content-Type': 'application/json',
+      },
+    });
+    const releaseData: ReleaseDataProps = res.data;
+    await Sleep(1000);
+    const obj = {
+      id: releaseData.id,
+      title: releaseData.title,
+      slug: releaseData.slug,
+      publicationId: releaseData.publicationId,
+      publicationTitle: releaseData.publicationTitle,
+      publicationSlug: releaseData.publicationSlug,
+      releaseName: releaseData.releaseName,
+      yearTitle: releaseData.yearTitle,
+      typeId: releaseData.typeId,
+      live: 'false',
+      timePeriodCoverage: {
+        value: releaseData.timePeriodCoverage.value,
+        label: releaseData.timePeriodCoverage.label,
+      },
+      latestRelease: 'false',
+      type: {
+        id: releaseData.type.id,
+        title: releaseData.type.title,
+      },
+      contact: {
+        id: releaseData.contact.id,
+        teamName: releaseData.contact.teamName,
+        teamEmail: releaseData.contact.teamEmail,
+        contactName: releaseData.contact.contactName,
+        contactTelNo: releaseData.contact.contactTelNo,
+      },
+      status: 'Approved',
+      amendment: 'false',
+      internalReleaseNote: 'Approved by publisher testing',
+      publishMethod: 'Immediate',
+    };
+    return obj;
+  } catch (e) {
+    console.error(e);
+  }
+  return;
+};
+
+const publishRelease = async (obj: any, releaseId: string) => {
+  try {
+    await axios({
+      method: 'PUT',
+      url: `${API_URL}/api/releases/${releaseId}`,
+      headers: {
+        Authorization: `Bearer ${JWT_TOKEN}`,
+        'Content-Type': 'application/json',
+      },
+      data: obj,
+    });
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+const getPublicationProgress = async (releaseId: string) => {
+  try {
+    const res = await axios({
+      maxContentLength: Infinity,
+      maxBodyLength: Infinity,
+      method: 'GET',
+      url: `${API_URL}/api/releases/${releaseId}/status`,
+      headers: {
+        Authorization: `Bearer ${JWT_TOKEN}`,
+        'content-Type': 'application/json',
+      },
+    });
+    return res.data?.overallStage
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+(async () => {
+  const zipGlob = await globby('*.zip');
+  if (fs.existsSync(zipGlob[0])) {
+    console.log('found zip file, continuing');
+  } else {
+    throw new Error(
+      'No zip file named archive.zip in root dir! Exiting test with failures',
+    );
+  }
+
+  rimraf(`./test-files/*`, () => {
+    console.log('cleaned test directory');
+  });
+
+  rimraf(`./zip-files/*`, () => {
+    console.log('cleaned zip file directory');
+  });
+
+  if (!fs.existsSync('./test-results')) {
+    fs.mkdirSync('./test-results');
+  }
+  if (!fs.existsSync('./test-files')) {
+    fs.mkdirSync('./test-files');
+  }
+
+  if (!fs.existsSync('./zip-files')) {
+    fs.mkdirSync('./zip-files');
+  }
+
+  const staleZipFiles = await globby('*.zip');
+  if (staleZipFiles[1]) {
```

Comment:
This checks that if there is a second zip file it removes it? But if globby found three zip files, then it would only remove one of them I believe? To prevent this problem you could put the zips the script creates into a separate directory so you can delete everything in that directory instead.

I think rather than rely on globby as much, it might be worth considering passing the zip file name/path to the script as an argument.

Reply:
There should never be an instance where there are 3 zip files in the project directory. Before sending the zip file to the API (with renamed data & meta files) it will move it into the root of the project (in order for the compiled index.js to pick it up). When you re run the test again the array returned from globby will look something like: 
```
[
'archive.zip', 
'clean-test-zip-<random_id>.zip'
]
```

So the test can safely assume that what is in staleZipFiles[1] is a stale zip file